lib/chef/provider/package/dpkg.rb

```diff
@@ -119,22 +97,65 @@ def upgrade_package(name, version)
         end
 
         def preseed_package(preseed_file)
-          Chef::Log.info("#{@new_resource} pre-seeding package installation instructions")
+          Chef::Log.info("#{new_resource} pre-seeding package installation instructions")
           run_noninteractive("debconf-set-selections #{preseed_file}")
         end
 
         def reconfig_package(name, version)
-          Chef::Log.info("#{@new_resource} reconfiguring")
+          Chef::Log.info("#{new_resource} reconfiguring")
           run_noninteractive("dpkg-reconfigure #{name}")
         end
 
+        private
+
+        def get_current_version
+          Chef::Log.debug("#{new_resource} checking install state")
+          status = shell_out_with_timeout("dpkg -s #{current_resource.package_name}")
+          package_installed = false
+          status.stdout.each_line do |line|
+            case line
+            when DPKG_INSTALLED
+              package_installed = true
+            when DPKG_VERSION
+              if package_installed
+                Chef::Log.debug("#{new_resource} current version is #{$1}")
+                return $1
+              end
+            end
+          end
+          return nil
+        end
+
         # Runs command via shell_out_with_timeout with magic environment to disable
         # interactive prompts. Command is run with default localization rather
         # than forcing locale to "C", so command output may not be stable.
-        #
-        # FIXME: This should be "LC_ALL" => "en_US.UTF-8" in order to stabilize the output and get UTF-8
         def run_noninteractive(command)
-          shell_out_with_timeout!(command, :env => { "DEBIAN_FRONTEND" => "noninteractive", "LC_ALL" => nil })
+          shell_out_with_timeout!(command, :env => { "DEBIAN_FRONTEND" => "noninteractive" })
```

Comment:
It seems to have been in response to this bug:

https://tickets.opscode.com/browse/CHEF-281

The root cause of that bug however was using the "en_US" locale which didn't exist on systems without en_US installed.  We've got complicated workarounds now which favor all kinds of different en_US locales and then fallback to to the 'C' locale as default.

The commit which introduced it just says:

Author: Diego Algorta <diego@oboxodo.com>
Date:   Fri Aug 14 19:54:31 2009 -0300

    Make sure all Package providers run with the system's locale and not C locale.
    
    Surely people want they're packages to corrently detect the system's
    locale when configuring themselves.

And I disagree completely with that logic.  If that logic was right, then if you installed a package with your machine set to en_US.UTF-8 and then changed your locale to de_DE.UTF-8 you would have to reinstall all your packages to get the "locale detection on package installation" right.  Packages can't have a requirement to 'detect' the locale on installation or else locale switching would be completely broken.